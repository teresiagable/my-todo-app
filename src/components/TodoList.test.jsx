import React from 'react';
import ReactDOM from 'react-dom';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TodoList, { TodoItem } from './TodoList';

configure({ adapter: new Adapter() });

const dummyData = [
  {
    id: 'Example-TodoItem-2',
    title: 'Create More Assignments!',
    description: 'Spawn new ideas with teammates.',
    deadline: '2020-01-02T08:00:00',
    done: false,
    assignee: {
      id: 'Example-User-3',
      firstName: 'Fredrik',
      lastName: 'Odin',
    },
  },
  {
    id: 'Example-TodoItem-3',
    title: 'Create More Exercises!',
    description: 'Spawn new ideas with teammates.',
    deadline: '2020-01-03T08:00:00',
    done: false,
    assignee: {
      id: 'Example-User-3',
      firstName: 'Fredrik',
      lastName: 'Odin',
    },
  },
];

it('displays 2 items', () => {
  const wrapper = shallow(<TodoList todolist={dummyData} />);
  const test = wrapper.find(TodoItem).length;
  expect(test).toEqual(2); // or the number of occurrence you're expecting
});

it('displays 0 items', () => {
  const wrapper = shallow(<TodoList todolist={[]} />);
  const test = wrapper.find(TodoItem).length;
  expect(test).toMatchSnapshot();
  expect(test).toEqual(0); // or the number of occurrence you're expecting
});
