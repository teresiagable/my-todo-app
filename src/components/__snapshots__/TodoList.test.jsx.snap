// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays 0 items 1`] = `0`;
